client/components/data/query-site-checklist/index.js

```diff
@@ -3,21 +3,26 @@
  */
 import PropTypes from 'prop-types';
 import { useEffect } from 'react';
-import { useDispatch } from 'react-redux';
+import { useDispatch, connect } from 'react-redux';
 
 /**
  * Internal dependencies
  */
 import { requestSiteChecklist } from 'state/checklist/actions';
+import isSiteUsingFullSiteEditing from 'state/selectors/is-site-using-full-site-editing';
 
-export default function QuerySiteChecklist( { siteId } ) {
+export function QuerySiteChecklist( { siteId, isSiteUsingFSE } ) {
 	const dispatch = useDispatch();
 
 	useEffect( () => {
-		dispatch( requestSiteChecklist( siteId ) );
-	}, [ dispatch, siteId ] );
+		dispatch( requestSiteChecklist( siteId, isSiteUsingFSE ) );
+	}, [ dispatch, siteId, isSiteUsingFSE ] );
 
 	return null;
 }
 
 QuerySiteChecklist.propTypes = { siteId: PropTypes.number.isRequired };
+
+export default connect( ( state, { siteId } ) => ( {
+	isSiteUsingFSE: isSiteUsingFullSiteEditing( state, siteId ),
+} ) )( QuerySiteChecklist );
```

Comment:
Using `connect` does the trick, but given this is already a functional component using the `useSelector` hook would remove an unnecessary layer in the component tree.

```javascript
import { useDispatch, useSelector } from 'react-redux';

// ...

export default function QuerySiteChecklist( { siteId } ) {
	const dispatch = useDispatch();
	const isSiteUsingFSE = useSelector(
		state => isSiteUsingFullSiteEditing( state, siteId )
	);

	useEffect( () => {
		dispatch( requestSiteChecklist( siteId, isSiteUsingFSE ) );
	}, [ dispatch, siteId, isSiteUsingFSE ] );

	return null;
}
```

Reply:
@p-jackson yeah, I tried swapping it out for `useSelector` but ran into an issue where it caused additional re-renders, whereas `connect` only renders the component once. I'm still pretty new to Redux hooks, so I wasn't sure the best way to ensure it only renders once?

Reply:
Of course the re-renders might not actually be a performance problem in this case 🤔

Reply:
TIL!

https://react-redux.js.org/next/api/hooks#performance
> unlike connect(), useSelector() does not prevent the component from re-rendering due to its parent re-rendering, even if the component's props did not change

And it doesn't look like there's a way around it either except to use `React.memo()` if performance is a problem. Which I'm guessing adds that extra level to the component tree again 😅 

Up to you then. I tend to ignore re-renders until I start profiling things based on a performance issue, but nothing wrong being pre-emptive either.

Reply:
Thanks! I'll take another look and see how I go :)